From b678265905159924233533337734b78073432662 Mon Sep 17 00:00:00 2001
From: Le Ma <le.ma@amd.com>
Date: Mon, 13 Jan 2020 12:26:00 +0800
Subject: [PATCH 1298/1453] drm/amdgpu: Temprarily disable vcn block for
 front-door loading method

VCN initialization will fail on part of Arcturus asics. This patch will
land in dkms branch to unblock sanity test.

Change-Id: I33cb10f30e7cd4f225fa202865c6ea59ede705d7
Signed-off-by: Le Ma <le.ma@amd.com>
Reviewed-by: Candice Li <Candice.Li@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/soc15.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/soc15.c b/drivers/gpu/drm/amd/amdgpu/soc15.c
index 6c75021dda0f..1588d4360a0e 100644
--- a/drivers/gpu/drm/amd/amdgpu/soc15.c
+++ b/drivers/gpu/drm/amd/amdgpu/soc15.c
@@ -772,7 +772,8 @@ int soc15_set_ip_blocks(struct amdgpu_device *adev)
 			if (likely(adev->firmware.load_type == AMDGPU_FW_LOAD_PSP))
 				amdgpu_device_ip_block_add(adev, &vcn_v2_5_ip_block);
 		} else {
-			amdgpu_device_ip_block_add(adev, &vcn_v2_5_ip_block);
+			if (likely(adev->firmware.load_type == AMDGPU_FW_LOAD_DIRECT))
+				amdgpu_device_ip_block_add(adev, &vcn_v2_5_ip_block);
 		}
 		if (!amdgpu_sriov_vf(adev))
 			amdgpu_device_ip_block_add(adev, &jpeg_v2_5_ip_block);
-- 
2.17.1

